File: user/src/main/java/com/dy/networkdisk/user/tool/QueueUtil.java
package com.dy.networkdisk.user.tool;

import org.apache.activemq.command.ActiveMQQueue;
import org.springframework.stereotype.Component;

import javax.annotation.PostConstruct;
import javax.jms.Queue;
import java.util.concurrent.ConcurrentHashMap;

@Component
public class QueueUtil {

    private ConcurrentHashMap<String,Queue> queueMap;

    @PostConstruct
    private void init(){
        queueMap = new ConcurrentHashMap<>();
    }

    public Queue get(String name){
        Queue queue = queueMap.get(name);
        if (queue == null){
            queue = new ActiveMQQueue(name);
            queueMap.put(name,queue);
        }
        return queue;
    }
}

File: user/src/main/kotlin/com/dy/networkdisk/user/tool/QueueUtil.kt
package com.dy.networkdisk.user.tool

import org.apache.activemq.command.ActiveMQQueue
import java.util.concurrent.ConcurrentHashMap
import javax.jms.Queue

class QueueUtil {
    companion object{
        private val queueMap = ConcurrentHashMap<String, Queue>()

        fun getQueue(name: String):Queue {
            return queueMap.getOrPut(name){
                ActiveMQQueue(name)
            }
        }
    }
}